server/consumer.go

```diff
@@ -795,6 +795,7 @@ func (mset *stream) addConsumerWithAssignment(config *ConsumerConfig, oname stri
 	}
 
 	// This is always true in single server mode.
+	o.mu.RLock()
```

Comment:
No, just isLeader. This is going to deadlock with o.sendCreateAdvisory...

Reply:
fixed